url.c

```diff
@@ -1455,23 +1458,23 @@ int url_mount(url_t *url, char *dir, int (*test_func)(url_t *))
   if(!url->is.network) {
     switch(url->scheme) {
       case inst_cdrom:
-        hw_item = hw_cdrom;
+        hw_items = (hd_hw_item_t[]) { hw_cdrom, 0 };
         break;
 
       case inst_floppy:
-        hw_item = hw_floppy;
+        hw_items = (hd_hw_item_t[]) { hw_floppy, 0 };
         break;
 
       default:
-        hw_item = hw_block;
+        hw_items = (hd_hw_item_t[]) { hw_block, 0 };
```

Comment:
NP: I see documentation that array have to be NULL ended, so maybe it makes sense to use constant here like `{ hw_block, NULL}` ?

Reply:
No, that's two different things. Here it's  just an array of enums, what you probably mean is the array of (hd_t *) (the detected  hardware structs).

Reply:
ah, now I see it. It is structure for hd_list2 method. Then fine for me.